OpenSL ES audio output

xyzz opened the pull request:
This adds the OpenSL ES audio output, used on Android.

There's an Android-specific header included, to my understanding opensl es isn't used anywhere else so this should be okay.

Also only two sound channels are supported at the moment, this is because I don't have any hardware to test more than two channels.

Here's the spec https://www.khronos.org/registry/sles/specs/OpenSL_ES_Specification_1.0.1.pdf. Note that android actually doesn't conform to this spec.

Tested on devices: Sony xperia z3 tablet compact running android 5.1.1 and nvidia shield tv running android 5.1

wm4 commented on audio/out/ao_opensles.c:
```diff
@@ -0,0 +1,230 @@
+    SLresult res;
+
+    double delay = p->buffer_size / (double)ao->bps;
+    ao_read_data(ao, data, p->buffer_size / ao->sstride, mp_time_us() + delay);
```
So I suppose there's still no reasonable latency/playback state estimation?

xyzz replied:
Yes, it doesn't look like the API has any function that does what we want.

wm4 commented on audio/out/ao_opensles.c:
```diff
@@ -0,0 +1,230 @@
+    double delay = p->buffer_size / (double)ao->bps;
+    ao_read_data(ao, data, p->buffer_size / ao->sstride, mp_time_us() + delay);
+
+    res = (*buffer_queue)->Enqueue(buffer_queue, p->buffer, p->buffer_size);
```
Hm, forces copying the data twice.

xyzz commented:
I've now removed the hack with Enqueue'ing 1 byte to get the buffer_callback to fire, and also switched to using 2 buffers, which should help to get lower latency on some devices.

wm4 commented:
Does it still have the jittering audio problem?

xyzz commented:
No improvements, the graph looks the same way to me: 
[stats_2buf.txt.gz](https://github.com/mpv-player/mpv/files/138192/stats_2buf.txt.gz)

wm4 commented:
So if this is in a useful state, I'll merge this?

xyzz commented:
Sure, go on.

wm4 commented:
Merged; slightly edited the commit message.
